sdk/common/src/main/scala/org/typelevel/otel4s/sdk/common/InstrumentationScope.scala

```diff
@@ -0,0 +1,173 @@
+/*
+ * Copyright 2023 Typelevel
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.typelevel.otel4s.sdk
+package common
+
+import cats.Hash
+import cats.Show
+import cats.syntax.show._
+
+/** Holds information about instrumentation scope.
+  *
+  * Instrumentation scope is a logical unit of the application code with which
+  * emitted telemetry is associated. The most common approach is to use the
+  * instrumentation library as the scope, however other scopes are also common,
+  * e.g. a module, a package, or a class may be chosen as the instrumentation
+  * scope.
+  *
+  * @see
+  *   [[https://opentelemetry.io/docs/specs/otel/glossary/#instrumentation-scope Instrumentation Scope spec]]
+  */
+sealed trait InstrumentationScope {
```

Comment:
I guess the Java SDK calls this `InstrumentationScopeInfo`, any commentary?

Reply:
The specification doesn't require this particular name. 
Java SDK uses `InstrumentationScopeInfo`, Go stands with [Scope](https://pkg.go.dev/go.opentelemetry.io/otel/sdk/instrumentation), Python uses [InstrumentationInfo](https://opentelemetry-python.readthedocs.io/en/latest/sdk/util.instrumentation.html) and Rust uses [InstrumentationLibrary](https://github.com/lzchen/opentelemetry-rust/blob/main/opentelemetry-api/src/common.rs#L441C12-L441C34).

From what I see, InstrumentationScope is sufficient enough and not verbose.

Reply:
> Python uses [InstrumentationInfo](https://opentelemetry-python.readthedocs.io/en/latest/sdk/util.instrumentation.html)

Wait a minute: Python has both an `InstrumentationInfo` and an `InstrumentationScope`. What's the difference? 🤔

Reply:
Perhaps they refer to [InstrumentationLibrary](https://opentelemetry.io/docs/specs/otel/glossary/#instrumentation-library) and [InstrumentationScope](https://opentelemetry.io/docs/specs/otel/glossary/#instrumentation-library)?

In JavaScript, there are both [InstrumentationLibrary](https://open-telemetry.github.io/opentelemetry-js/interfaces/_opentelemetry_core.InstrumentationLibrary.html) and [InstrumetnationScope](https://open-telemetry.github.io/opentelemetry-js/interfaces/_opentelemetry_core.InstrumentationScope.html). InstrumentationLibrary is deprecated, though.